appender_test.go

```diff
@@ -720,8 +722,9 @@ func TestAppenderFlushRequestError(t *testing.T) {
 		))
 
 		indexer, err := docappender.New(client, docappender.Config{
-			FlushInterval: time.Minute,
-			MeterProvider: sdkmetric.NewMeterProvider(sdkmetric.WithReader(rdr)),
+			FlushInterval:        time.Minute,
+			MeterProvider:        sdkmetric.NewMeterProvider(sdkmetric.WithReader(rdr)),
+			IncludeSourceOnError: docappender.True,
```

Comment:
As mentioned below, the Unset, False, True handling (as opposed to propagation) should be tested as well. False and True should pass through the error, Unset should sanitize reason.